cvat-ui/src/components/annotation-page/standard-workspace/controls-side-bar/opencv-control.tsx

```diff
@@ -215,6 +269,45 @@ class OpenCVControlComponent extends React.PureComponent<Props & DispatchToProps
         return points;
     }
 
+    private imageModifier(alias: string): ImageProcessing|null {
+        const { activeImageModifiers } = this.state;
+        return activeImageModifiers.find((imageModifier) => imageModifier.alias === alias)?.modifier || null;
+    }
+
+    private disableImageModifier(alias: string):void {
+        const { activeImageModifiers } = this.state;
+        const index = activeImageModifiers.findIndex((imageModifier) => imageModifier.alias === alias);
+        if (index !== -1) {
+            activeImageModifiers.splice(index, 1);
+            this.setState({
+                activeImageModifiers: [...activeImageModifiers],
+            });
+        }
+    }
+
+    private enableImageModifier(modifier: ImageProcessing, alias: string): void{
+        const { activeImageModifiers } = this.state;
+        activeImageModifiers.push({ modifier, alias });
+        this.runImageModifier();
+        this.setState({
+            activeImageModifiers: [...activeImageModifiers],
+        });
```

Comment:
Both runImageMidifers and setState are async, so, more safe option is:

```ts
this.setState((prev: State) => ({
        ...prev,
        activeImageModifiers: [...prev.activeImageModifiers, { modifier, alias }]
}), () => {
      this.runImageModifier();
});
```

What do you think?

Reply:
Seems reasonable. Its unsafe to run them one after another